dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataSqlQueryFactoryBuilder.java

```diff
@@ -0,0 +1,288 @@
+        "L.NumFragments",
+        "L.NumOfActiveAMPs",
+        "L.NumRequestCtx",
+        "L.NumResultOneMBRows",
+        "L.NumResultRows",
+        "L.NumSteps",
+        "L.NumStepswPar",
+        "L.ParamQuery",
+        "L.ParserCPUTime",
+        "L.ParserCPUTimeNorm",
+        "L.ParserExpReq",
+        "L.PersistentSpool",
+        "L.ProcID",
+        "L.ProfileID",
+        "L.ProfileName",
+        "L.ProxyRole",
+        "L.ProxyUser",
+        "L.ProxyUserID",
+        "L.QueryBand",
+        "L.QueryRedriven",
+        "L.QueryText",
+        "L.ReDriveKind",
+        "L.RemoteQuery",
+        "L.ReqIOKB",
+        "L.ReqPhysIO",
+        "L.ReqPhysIOKB",
+        "L.RequestMode",
+        "L.RequestNum",
+        "L.SeqRespTime",
+        "L.SessionID",
+        "L.SessionTemporalQualifier",
+        "L.SpoolUsage",
+        "L.StartTime AT TIME ZONE INTERVAL '0:00' HOUR TO MINUTE AS \"StartTime\"",
+        "L.StatementGroup",
+        "L.Statements",
+        "L.StatementType",
+        "L.SysDefNumMapAMPs",
+        "L.TacticalCPUException",
+        "L.TacticalIOException",
+        "L.TDWMEstMemUsage",
+        "L.ThrottleBypassed",
+        "L.TotalFirstRespTime",
+        "L.TotalIOCount",
+        "L.TotalServerByteCount",
+        "L.TTGranularity",
+        "L.TxnMode",
+        "L.TxnUniq",
+        "L.UnitySQL",
+        "L.UnityTime",
+        "L.UsedIota",
+        "L.UserID",
+        "L.UserName",
+        "L.UtilityByteCount",
+        "L.UtilityInfoAvailable",
+        "L.UtilityRowCount",
+        "L.VHLogicalIO",
+        "L.VHLogicalIOKB",
+        "L.VHPhysIO",
+        "L.VHPhysIOKB",
+        "L.WarningOnly",
+        "L.WDName"
+      };
+
+  private ZonedInterval interval;
+  private String logTable;
+  private String queryTable;
+  private List<String> conditions;
+  private List<String> orderColumns;
+  private List<String> expressions;
+  private boolean isAssessment;
+  // For Terdata version <= 14
+  private boolean isOldVersion;
+  private final boolean isLSql;
+
+  /* pp */ static TeradataSqlQueryFactoryBuilder startBuildingFrom(
+      ConnectorArguments arguments, boolean forLSql) throws MetadataDumperUsageException {
+    TeradataSqlQueryFactoryBuilder builder = new TeradataSqlQueryFactoryBuilder(forLSql);
+    builder.parse(arguments);
+    return builder;
+  }
+
+  /* pp */ static TeradataSqlQueryFactoryBuilder startBuildingFrom(ConnectorArguments arguments)
+      throws MetadataDumperUsageException {
+    return startBuildingFrom(arguments, false);
+  }
+
+  /* pp */ TeradataSqlQueryFactoryBuilder within(ZonedInterval interval) {
+    this.interval = interval;
+    return this;
+  }
+
+  private TeradataSqlQueryFactoryBuilder(boolean isLSql) {
+    this.isLSql = isLSql;
+  }
+
+  private void parse(ConnectorArguments arguments) throws MetadataDumperUsageException {
+    isAssessment = arguments.isAssessment();
+    isOldVersion = arguments.getConnectorName().startsWith("teradata14");
```

Comment:
nit: Consider using other name for this variable, e.g. `isTeradata14Version`, as current can be misinterpreted as the old version of the dumper.